### DISCUSSION TITLE
Themes for a Decoupled Multi Tenant Site?

### DISCUSSION THREAD

**Discussion by selaromdotnet** [OrchardCMS/OrchardCore#8676] [Q&A]
hello all! I'm crossposting this from a chat as I originally mistakenly thought I fixed the problem, but turns out either I was wrong or I broke it again and I don't know if I'm also just crazy...

so just started playing with orchard core, and i'm having a tough time discovering an answer to this question: When using OC for a decoupled site can you use the OC theme projects to create themes for the decoupled front end site? 

I created two tenants in a decoupled site, blank site a (default) and blank site b, then created a new OC theme, adding a reference of it to the main project. This site has a single page Index.cshtml and a _Layout.cshtml in the Shared folder.

in the theme all I added was a variant of the Pages/Shared/_Layout.cshtml file. 

if I login to site b, I can see and select the theme, but it doesn't appear to do anything to the front end; it's still serving the main _Layout file from the root site. 

if I delete this file from the root site, one of the paths it searches for a replacement is "/Areas/SiteA/Pages/Shared/_Layout.cshtml" which looks close to what i expect, but this is not an area, it's in the separate theme project. 

Am I attempting to do something that is not supported, or am I just missing a step somewhere?

Many thanks!

**Comment by deanmarcussen** [comment]
I think if you're wanting to use themeing like this, you will need two themes, one for each.

You can still use decoupled approaches within themes. 

You just need to place the appropriate views in the themes.

The main host acts as a supertheme so will always override any views placed in it.

**Reply by selaromdotnet** [reply]
thank you for the reply, I appreciate the insight!

But isn't this the opposite of what should happen? I thought it resolved themes inward, so that placing a layout file in the theme would override the lower layers...

even if that's true, wouldn't deleting the root layout let the theme layout override? it is saying the path isn't found, so i'm thinking maybe I'm doing something wrong on the  build or reference that is preventing it from locating the external resource...

any ideas how i can troubleshoot this?

**Reply by jtkech** [reply]
@selaromdotnet 

All view templates (razor or liquid), including the layout used by the theming engine, are harvested from the `Views` folders of the related modules / themes (including the app), not from the `Pages` folders. Razor view (`Razor.RazorPage`) can be used as shape view template or pure mvc views, Razor pages (`RazorPages.Page`) are only used in the mvc way.

**Reply by selaromdotnet** [reply]
ok thanks again, I do recall when I moved stuff out of the Pages folder in the theme into the Views folder coincided it with working briefly...

but when I tried to do it with a real theme it just stopped working.

This is what my setup looks like:

![image](https://user-images.githubusercontent.com/566534/109358949-3bbbab80-784a-11eb-8b42-e79ce5bd576a.png)

These are three different layout files, and in the _ViewStart.cshtml of the main site it just points to "Layout".

The root site has no theme, and loads the root site's layout. the other two tenants I created (Bar and Restaurant) have each theme respectively assigned, but all three sites load the same root layout.

I tried removing Viewstart and putting Layout directly in the Index.cshtml page but that didn't work, neither did assigning a different theme to the main site. no matter what I do I get the root site layout.

I swear I got this to work once, but I can't understand what I changed to break it, or how to get it back to a working state.

I was definitely using pages when it worked (not controller/view), but that's about as far as I can get...

appreciate any other insight that might be helfpul, thanks!

**Reply by selaromdotnet** [reply]
ahhhh now i know what I did: the files in the themes need to be configured as EmbeddedResource. Making this change made everything popup  and work as expected!!!

I couldn't find any mention of this anywhere in any of the docs, seems obvious now, but I think it's worth mentioning. I'd say I can't be the only one this happened to, but I certainly didn't have any luck finding any info about this anywhere!

anyway, finally back to being productive, thanks again for the help!

**Comment by remesq** [comment]
Not sure if this would help https://github.com/psijkof/ModernBusiness.OC.RazorPages. It's a bit older now, but I use it for concepts with razor pages and decoupled sites.
